### PR TITLE
Add All/Remove All buttons for user groups

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
initial view
screen shot 2018-06-01 at 2 26 58 pm

demonstrating works with filters
screen shot 2018-06-01 at 2 27 10 pm
screen shot 2018-06-01 at 2 27 16 pm

### CODE REVIEW

**Comment by esotrope** [comment]
I like the positioning of the buttons.
I think it might look better if we make them links without underlines though. Also we should be consistent with using either `x` or `-` to represent remove.

File: webapp/src/main/webapp/src/app/user-group/user-group-form.component.html [hunk 1]
```diff
@@ -24,7 +24,19 @@
...
   <div class="form-group">
-    <label>{{'user-group-form.field.students.label' | translate}}</label>
+    <div class="flex-children">
+      <div class="flex-child grow pb-sm">
+        <label>{{'user-group-form.field.students.label' | translate}}</label>
+      </div>
+      <div *ngIf="group.students.length" class="flex-child">
+        <button type="submit"
+                class="btn btn-danger btn-sm"
+                (click)="removeAllStudents()">
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/user-group/user-group-form.component.html]: I would give some indirection here like `(click)="removeAllStudentsButtonClick()` also why is the button `type=submit`?

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/user-group/user-group-form.component.html]: Sounds good. Not sure (copy and pasted). Should it be `type="button"`?

File: webapp/src/main/webapp/src/app/user-group/user-group.component.ts [hunk 2]
```diff
@@ -195,16 +195,24 @@ export class UserGroupComponent implements OnInit, OnDestroy {
...
   onFormStudentClick(student: Student): void {
+    this.addStudents(student);
+  }
+
+  addAllStudents() {
+    this.addStudents(...this.filteredStudents);
+  }
+
+  private addStudents(...student: Student[]) {
+    const studentsToAdd = student.filter(aStudent => !this.group.students.includes(aStudent));
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/user-group/user-group.component.ts]: not sure this is needed. we filter the student search results already based on those in the group already

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/user-group/user-group.component.ts]: ah gotcha. I was passing along `this.students` while I was doing this initially until I realized that would be incorrect. I'll remove this

File: webapp/src/main/webapp/src/app/user-group/user-group-form.component.html (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/user-group/user-group-form.component.html]: yeah you can either omit it or make it type=button

**Pull request merged**